framework/arcane-framework/src/main/scala/services/lakehouse/IcebergS3CatalogWriter.scala

```diff
@@ -106,6 +104,21 @@ class IcebergS3CatalogWriter(
     Future(catalog.loadTable(tableId)).flatMap(appendData(data, schema, false))
 
 object IcebergS3CatalogWriter:
+  /**
+   * The ZLayer that creates the LazyOutputDataProcessor.
+   */
+  val layer: ZLayer[IcebergCatalogSettings, Nothing, CatalogWriter[RESTCatalog, Table, Schema]] =
```

Comment:
i'd say all ZIO stuff should sit in its own object, since this guy is meant to be pure Iceberg on S3

Reply:
Discussed by voice: the `layer` property is tightly coupled with class **implementation** not the interface and cannot be generalized. Additionally, placing the `layer` property in the same object as a constructor follows the change locality principle and idiomatic (see [zio docs](https://zio.dev/reference/di/dependency-injection-in-zio/)).

This variable does not provide any dependency on ZIO or side effects for caller if caller does not use ZIO.
So we agreed that it's ok to leave this object as is.